hmm i think this is actually failing see the traivs CI: https://travis-ci.org/github/pandas-dev/pandas/jobs/739214922

```
>   ranked_mat[:, i] = rank_1d(mat[:, i])
E   TypeError: Argument 'in_arr' has incorrect type (expected numpy.ndarray, got pandas._libs.algos._memoryviewslice)
pandas/_libs/algos.pyx:347: TypeError
_ TestDataFrameCorr.test_corr_nullable_integer[spearman-other_column2-nullable_column1] _
[gw0] linux -- Python 3.7.9 /home/travis/miniconda3/envs/pandas-dev/bin/python
self = <pandas.tests.frame.methods.test_cov_corr.TestDataFrameCorr object at 0x7fcbc4f5a4d0>
nullable_column = <IntegerArray>
[1, 2, <NA>]
Length: 3, dtype: Int64
other_column = array([ 1.,  2., nan]), method = 'spearman'
    @td.skip_if_no_scipy
    @pytest.mark.parametrize(
        "nullable_column", [pd.array([1, 2, 3]), pd.array([1, 2, None])]
    )
    @pytest.mark.parametrize(
        "other_column",
        [pd.array([1, 2, 3]), np.array([1.0, 2.0, 3.0]), np.array([1.0, 2.0, np.nan])],
    )
    @pytest.mark.parametrize("method", ["pearson", "spearman", "kendall"])
    def test_corr_nullable_integer(self, nullable_column, other_column, method):
        # https://github.com/pandas-dev/pandas/issues/33803
        data = DataFrame({"a": nullable_column, "b": other_column})
>       result = data.corr(method=method)
pandas/tests/frame/methods/test_cov_corr.py:190: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/frame.py:8300: in corr
    correl = libalgos.nancorr_spearman(mat, minp=min_periods)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ranked_mat[:, i] = rank_1d(mat[:, i])
E   TypeError: Argument 'in_arr' has incorrect type (expected numpy.ndarray, got pandas._libs.algos._memoryviewslice)
```

--- a/pandas/_libs/algos.pyx
+++ b/pandas/_libs/algos.pyx
@@ -325,7 +325,7 @@ def nancorr(const float64_t[:, :] mat, bint cov=False, minp=None):
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def nancorr_spearman(const float64_t[:, :] mat, Py_ssize_t minp=1) -> ndarray:
+def nancorr_spearman(ndarray[float64_t, ndim=2] mat, Py_ssize_t minp=1) -> ndarray:
     cdef:
         Py_ssize_t i, j, xi, yi, N, K
         ndarray[float64_t, ndim=2] result
@@ -1018,7 +1018,7 @@ def rank_1d(
 
 
 def rank_2d(
-    rank_t[:, :] in_arr,
+    ndarray[rank_t, ndim=2] in_arr,
     int axis=0,
     ties_method="average",
     bint ascending=True,

